src/adapter/src/coord/sequencer.rs

```diff
@@ -1255,40 +1268,22 @@ impl<S: Append + 'static> Coordinator<S> {
         } = plan;
 
         // First try to allocate an ID and an OID. If either fails, we're done.
-        let id = match self.catalog.allocate_user_id().await {
-            Ok(id) => id,
-            Err(e) => {
-                tx.send(Err(e.into()), session);
-                return;
-            }
-        };
-        let oid = match self.catalog.allocate_oid() {
-            Ok(id) => id,
-            Err(e) => {
-                tx.send(Err(e.into()), session);
-                return;
-            }
-        };
+        let id = try_exec!(self.catalog.allocate_user_id().await, tx, session);
```

Comment:
A thought: if we ever need more than one call to this macro in a function, then the function should be split into two where one part can return an error, and the other part does the tx sending correctly.

Reply:
Good idea, I modified peek and explain timestamp for this.